### DISCUSSION TITLE
How to get related collection in custom interface?

### DISCUSSION THREAD

**Discussion by IlyaPetrovM** [directus/directus#12979] [Archive]
I want to create custom o2m Interface. 

What should i mention in props to get related values?

**Comment by rijkvanzanten** [comment]
If you mark the interface as a relational interface in it's configuration object, the app will present the relational configuration when using this field:

https://github.com/directus/directus/blob/0f27290fd11a8a26ceb8605d3dddc5d37b7f7cb8/app/src/interfaces/list-o2m/index.ts#L11-L14

**Reply by IlyaPetrovM** [reply]
But how it will present relational configuration?

I wrote this config, but I dont know what to write in interface props and template.

**index.js**
```js
import InterfaceComponent from './interface.vue';

export default {
        id: 'table_with_links',
        name: 'Table with links',
        icon: 'arrow_right_alt',
        component: InterfaceComponent,
        types: ['alias'],
        localTypes: ['o2m'],
        group: 'relational',
        relational: true,
};
```

**interface.vue**
```vue
<template>
        <v-list>
                <v-list-item v-for='item in value'> {{ item.title }} </v-list-item>
        </v-list>
</template>

<script>
import { useApi } from '@directus/extensions-sdk';

export default {
        props: {
                value: {
                        type: Object,
                        default: null,
                },
                collection: {
                        type: String,
                },
                primaryKey: {
                        type: String | Number,
                },
                field: {
                        type: String,
                },

                token: String,
                template: String,
        },
        emits: ['input'],
        setup(props, { emit }) {
                const api = useApi();
                props.token = api.defaults.headers.common['Authorization']?.split(' ')[1] || null;
                console.log(props);
        },
};
</script>

**Reply by rijkvanzanten** [reply]
The relationship information isn't passed down. You should be able to extract that information from the relationships store:

```js
import { useStores } from "@directus/extensions-sdk";

const { useRelationsStore } = useStores();

const relationsStore = useRelationsStore();
const relations = relationsStore.getRelationsForField(props.collection, props.field);
```

**Reply by IlyaPetrovM** [reply]
Thank you! 
Now everything is clear!
 
I'm sorry for these stupid questions. The fact is that I have not found a description of the relationsStore methods in docs.

**Reply by rijkvanzanten** [reply]
No worries! These things aren't obvious, and often not (properly) documented (yet)! @br41nslug is working on getting some auto generated docs going for these "internal" APIs for use in extensions in https://github.com/directus/directus/pull/12593, so hopefully there's more good news on that soon 🙂
